crates/figma_import/src/toolkit_schema.rs

```diff
@@ -19,13 +19,55 @@ use std::sync::atomic::AtomicU16;
 // retain image references.
 use serde::{Deserialize, Serialize};
 
-use crate::figma_schema::Rectangle;
+use crate::color::Color;
+use crate::figma_schema;
 use crate::reaction_schema::FrameExtras;
 use crate::reaction_schema::Reaction;
 use crate::toolkit_style::{StyledTextRun, ViewStyle};
+use std::collections::HashMap;
 
-pub use crate::figma_schema::OverflowDirection;
-pub use crate::figma_schema::StrokeCap;
+pub use crate::figma_schema::{FigmaColor, OverflowDirection, Rectangle, StrokeCap, VariableAlias};
+
+// Enum for fields that represent either a fixed number or a number variable
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub enum NumOrVar {
+    Num(f32),
+    Var(String),
+}
+impl NumOrVar {
+    pub(crate) fn from_var(
+        bound_variables: &figma_schema::BoundVariables,
+        var_name: &str,
+        num: f32,
+    ) -> NumOrVar {
+        let var = bound_variables.get_variable(var_name);
+        if let Some(var) = var {
+            NumOrVar::Var(var)
+        } else {
+            NumOrVar::Num(num)
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub enum ColorOrVar {
+    Color(Color),
+    Var(String),
+}
+impl ColorOrVar {
+    pub(crate) fn from_var(
+        bound_variables: &figma_schema::BoundVariables,
+        var_name: &str,
+        color: &FigmaColor,
+    ) -> ColorOrVar {
+        let var = bound_variables.get_variable(var_name);
+        if let Some(var) = var {
+            ColorOrVar::Var(var)
```

Comment:
This is largely duplicated code. Can we make a generic Variable<T> type?

Reply:
The idea of the "XorVar" types is that Figma nodes that use variables are typically either set to a fixed value, such as the color #FF0000, or a variable ID, such as "VariableId:12345". Hence I came up with "ColorOrVar", since the background of the node is one of the two values in the enum. So Figma nodes that use variables use these "XorVar" values, whereas the large variable table that holds all variables in a Figma file use the `Variable` struct below.

Reply:
Makes sense to keep the things that are set separate from the entire list, but I was specifically referring to creating a generic XorVar<T> type so that we don't have four sets of largely duplicated code.

Reply:
I looked into this but not sure if that would really be worth it. I could make something like:
```
pub enum BoundVariable {
    Num(f32),
    Color(Color),
    Var(String),
}
```
to encapsulate all the different variable types, but I'd still have to make separate functions `from_num_var()`, `from_color_var()`, etc. to initialize them from different types of bound variable data. So the code savings would be minimal. Plus by keeping it the way it is, we know that a background color is a `ColorOrVar`, which is either a color value or a variable, not just a more generic `BoundVariable` that can be any type.

Reply:
An `XorVar<T>` (or `BoundVariable<T>`) type wouldn't work? That's what I meant by [generic](https://google.github.io/comprehensive-rust/generics/generic-data.html).

Reply:
I looked into this, and although it is doable, having a generic struct/enum makes serde reflection harder. In order to generate XorVar<f32>, XorVar<Color>, etc, I would need to pull in the serde-name crate, then override the serialize and deserialize functions to generate different named classes for each different generic parameter. At this point I don't think it's worth the work. Plus, this might change during your protobuf work.

Reply:
Ok, yeah, we'll look at it again with the Protobuf work. I'd been using the pattern of having individual messages (such as `DimensionRect`) and doing some simple mapping to the generic Rust native structs (`Rect<Dimension>`) to handle generics and I think that'll work here too.